### PR TITLE
Improved Error Handling and Other Improvements

### PR DESCRIPTION
**Pull request by emurphy-9**
Hey, I know it's been a while since I first raised the issue (guess I was busier than I thought :sweat_smile:  ). I added in the suggestions if the user fails to select a valid function name.

... I also got kind of carried away :sweat_smile: and made some changes to the structure of the code, and made use of some of the latest Javascript features (like the spread operator). I did add tests for everything though... And also changed the structure of the test files. It should be clean and easy to read. :smiley:

### CODE REVIEW

**Comment by LasaleFamine** [comment]
Hey there! This is actually more than expected, awesome work! 💯 
Just give me some time to have a look at the changes, but I think we are pretty ok. 

Maybe another check should be done for the coverage, I also wanted to switch from coveralls to [Codecov](https://codecov.io/) since it is more accurate, but I can take a look at this later.

Thank you again, I'll be back soon 👍

**Comment by emurphy-9** [comment]
Yeah the coverage is failing because of some dummy functions I added to the tests (that technically don't get executed, so uncovered lines). So that could probably be configured to ignore the test file that triggers it. 

Sure take your time, there's a fair few changes to go through, and I'm happy to help/explain anything crazy I did. :smile:

File: README.md [hunk 1]
```diff
@@ -1,9 +1,8 @@
 # runnable-exports
-[![Build Status](https://travis-ci.org/LasaleFamine/runnable-exports.svg?branch=master)](https://travis-ci.org/LasaleFamine/runnable-exports) [![Coverage Status](https://coveralls.io/repos/github/LasaleFamine/runnable-exports/badge.svg)](https://coveralls.io/github/LasaleFamine/runnable-exports)
-
+[![Build Status](https://travis-ci.com/emurphy-9/runnable-exports.svg?branch=master)](https://travis-ci.com/emurphy-9/runnable-exports) [![Coverage Status](https://coveralls.io/repos/github/emurphy-9/runnable-exports/badge.svg?branch=master)](https://coveralls.io/github/emurphy-9/runnable-exports?branch=master)
```
Comment (LasaleFamine) [inline README.md]: I think this should be reverted 😄

**Review by LasaleFamine — CHANGES_REQUESTED** [review]
Really awesome work! We just need to revert the badge and I think we can merge it.

**Comment by emurphy-9** [comment]
Oh yeah, I forgot about that 😅 Thanks! 😊

File: README.md (hunk 1 above)
Comment (LasaleFamine) [reply README.md]: I'll take care of this 🤙 I will close the related issue #1 when released. Thanks a lot again!

**Review by LasaleFamine — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (emurphy-9) [reply README.md]: Awesome! No problem, it was my pleasure. Thanks for taking the time to respond and merge everything in! 🙂

**Pull request merged**